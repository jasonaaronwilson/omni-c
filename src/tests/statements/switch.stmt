switch(i) {
  case 0: break;
  default: return 10;
}
