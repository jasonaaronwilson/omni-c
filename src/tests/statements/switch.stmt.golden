{
    tag = PARSE_NODE_SWITCH_STATEMENT
    expression = {
        tag = PARSE_NODE_IDENTIFIER
        token = i
    }
    block = {
        tag = PARSE_NODE_BLOCK
        statements = [
            {
                tag = PARSE_NODE_CASE_LABEL
                expression = {
                    tag = PARSE_NODE_LITERAL
                    tokens = [
                        0
                    ]
                }
            }
            {
                tag = PARSE_NODE_BREAK_STATEMENT
                break_keyword_token = break
            }
            {
                tag = PARSE_NODE_DEFAULT_LABEL
            }
            {
                tag = PARSE_NODE_RETURN_STATEMENT
                expression = {
                    tag = PARSE_NODE_LITERAL
                    tokens = [
                        10
                    ]
                }
            }
        ]
    }
}

// C Output
switch (i)
{
  case 0:
  break;
  default:
  return 10;
}

